spec/validation/meta_yml_spec.rb

```diff
@@ -1,103 +1,140 @@
 # frozen_string_literal: true
 require "spec_helper"
+require "pry"
 
-module HathiTrust
-  module Validation
+include HathiTrust
+include Validation::MetaYml
 
-    shared_examples_for "a validator with a valid package" do
-      it "does not return errors" do
-        expect(any_errors?(validator.validate)).to be_falsey
-      end
+shared_examples_for "a validator with a valid package" do
+  it "does not return errors" do
+    expect(any_errors?(validator.validate)).to be_falsey
+  end
+end
+
+shared_examples_for "a validator with an invalid package" do
+  it "returns a collection of Validation::Messages" do
+    expect(validator.validate).to all(be_an_instance_of(Validation::Message))
+  end
+end
+
+shared_examples_for "a validator with warnings and only warnings" do
+  it "has at least one warning" do
+    expect(validator.validate.any?(&:warning?)).to be_truthy
+  end
+
+  it "has only warnings" do
+    expect(validator.validate.all?(&:warning?)).to be_truthy
+  end
+end
+
+def valid_yaml
```

Comment:
It can. Generally I was trying to avoid `let` unless it simplified things over a helper function, but I think for this case `let` probably makes sense. The "Testing Rails" book (https://robots.thoughtbot.com/announcing-testing-rails) thinks `let` and some things like `before(:each)` are an example of the "Mystery Guest" anti-pattern. I can see their point in some complicated cases; I don't think this is one of them, so I think `let` is fine here. Shared contexts with `let` are perhaps a better demonstration of the issue as it can make it a little bit tricky to find out where something is defined, but I think it's fine.